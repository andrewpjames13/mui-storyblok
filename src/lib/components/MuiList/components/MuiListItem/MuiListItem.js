import React from 'react';
import PropTypes from 'prop-types';
import ListItem from '@material-ui/core/ListItem';
import {
  validComponents,
  muiStringProp,
  validComponentsRequired,
} from '../../../../utils/customProps';
import Storyblok from '../../../../utils/Storyblok';
import MuiListItemAvatar from './components/MuiListItemAvatar/MuiListItemAvatar';
import MuiListItemIcon from './components/MuiListItemIcon/MuiListItemIcon';
import MuiListItemSecondaryAction from './components/MuiListItemSecondaryAction/MuiListItemSecondaryAction';
import MuiListItemText from './components/MuiListItemText/MuiListItemText';
import MuiContactButton from '../../../MuiContactButton/MuiContactButton';

const MuiListItem = ({
  rootClass,
  alignItems,
  dense,
  disableGutters,
  divider,
  selected,
  contactButton,
  listItemAvatar,
  listItemIcon,
  listItemSecondaryAction,
  listItemText,
}) => {
  const styles = Storyblok.arrayToMuiStyles(rootClass);

  const avatar = listItemAvatar[0];
  const icon = listItemIcon[0];
  const secondaryAction = listItemSecondaryAction[0];
  const text = listItemText[0];
  const contact = contactButton[0];
  return (
    <ListItem
      className={styles.root}
      dense={dense}
      alignItems={alignItems}
      disableGutters={disableGutters}
      divider={divider}
      selected={selected}
    >
      {avatar ? <MuiListItemAvatar {...avatar} /> : null}
      {icon ? <MuiListItemIcon {...icon} /> : null}
      {text ? <MuiListItemText {...text} /> : null}
      {secondaryAction ? <MuiListItemSecondaryAction {...secondaryAction} /> : null}
      {contact ? <MuiContactButton {...contact} /> : null}
    </ListItem>
  );
};

export default MuiListItem;

MuiListItem.propTypes = {
  /** stroyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /** mui prop: 'flex-start', 'center'
   * Defines the align-items style property.
  */
  alignItems(props, propName, componentName) {
    const validProps = ['flex-start', 'center'];
    return muiStringProp(props, propName, componentName, validProps);
  },
  /** mui prop: true | false
   * If true, compact vertical padding designed for keyboard and mouse input will be used.
   * */
  dense: PropTypes.bool,
  /** mui prop: true | false
   * If true, the left and right padding is removed.
   * */
  disableGutters: PropTypes.bool,
  /** mui prop: true | false
   * If true, a 1px light border is added to the bottom of the list item.
   * */
  divider: PropTypes.bool,
  /** mui prop: true | false
   * Use to apply selected styling.
   * */
  selected: PropTypes.bool,
  /** MuiListItemAvatar Allowed maximum: 1 */
  listItemAvatar(props, propName, componentName) {
    const components = ['MuiListItemAvatar'];
    return validComponents(props, propName, componentName, components, 1);
  },
  /** MuiListItemAvatar Allowed maximum: 1 */
  contactButton(props, propName, componentName) {
    const components = ['MuiListItemAvatar'];
    return validComponents(props, propName, componentName, components, 1);
  },
  /** MuiListItemIcon Allowed maximum: 1 */
  listItemIcon(props, propName, componentName) {
    const components = ['MuiListItemIcon'];
    return validComponents(props, propName, componentName, components, 1);
  },
  /** MuiListItemText Allowed maximum: 1 */
  listItemText(props, propName, componentName) {
    return validComponentsRequired(props, propName, componentName, ['MuiListItemText'], 1);
  },
  /** MuiListItemSecondaryAction Allowed maximum: 1 */
  listItemSecondaryAction(props, propName, componentName) {
    const components = ['MuiListItemSecondaryAction'];
    return validComponents(props, propName, componentName, components, 1);
  },
};

MuiListItem.defaultProps = {
  alignItems: 'center',
  dense: false,
  disableGutters: false,
  divider: false,
  selected: false,
  rootClass: [],
  listItemAvatar: [],
  listItemIcon: [],
  listItemSecondaryAction: [],
  contactButton: [],
  listItemText: [],
};
