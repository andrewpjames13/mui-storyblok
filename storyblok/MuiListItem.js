module.exports = {
  "name": "MuiListItem",
  "display_name": null,
  "updated_at": "2020-05-06T14:54:06.784Z",
  "schema": {
    "listItemText": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiListItemText"
      ],
      "maximum": "1",
      "required": true,
      "pos": 0
    },
    "rootClass": {
      "type": "options",
      "pos": 1
    },
    "alignItems": {
      "type": "option",
      "use_uuid": true,
      "options": [
        {
          "value": "flex-start",
          "name": "flex-start"
        },
        {
          "value": "center",
          "name": "center"
        }
      ],
      "default_value": "center",
      "pos": 2
    },
    "dense": {
      "type": "boolean",
      "pos": 3
    },
    "disableGutters": {
      "type": "boolean",
      "pos": 4
    },
    "divider": {
      "type": "boolean",
      "pos": 5
    },
    "selected": {
      "type": "boolean",
      "pos": 6
    },
    "listItemAvatar": {
      "type": "bloks",
      "restrict_components": true,
      "maximum": "1",
      "component_whitelist": [
        "MuiListItemAvatar"
      ],
      "pos": 7
    },
    "listItemIcon": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiListItemIcon"
      ],
      "maximum": "1",
      "pos": 8
    },
    "listItemSecondaryAction": {
      "type": "bloks",
      "maximum": "1",
      "restrict_components": true,
      "component_whitelist": [
        "MuiListItemSecondaryAction"
      ]
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiListItem",
  "component_group_uuid": null
}