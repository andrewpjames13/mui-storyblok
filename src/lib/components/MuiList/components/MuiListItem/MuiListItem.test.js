import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { MuiListItem } from './MuiListItem';

function setup(isButton = false, redirectRoute = undefined) {
  const props = {
    listItemAvatar: [{
      component: 'MuiListItemAvatar',
      content: [{
        component: 'MuiIcon',
        iconName: 'android',
      }],
    }],

    listItemIcon: [{
      component: 'MuiListItemIcon',
      content: [{
        component: 'MuiIcon',
        iconName: 'android',
      }],
    }],

    listItemText: [{
      component: 'MuiListItemText',
      primary: 'primary',
    }],

    listItemSecondaryAction: [{
      component: 'MuiListItemSecondaryAction',
      content: [{
        component: 'MuiIconButtonRedirect',
        iconButton: [{
          component: 'MuiIconButton',
          icon: [{
            component: 'MuiIcon',
            iconName: 'star',
          }],
        }],
      }],
    }],
    isButton,
    redirectRoute,
    history: {
      push: jest.fn(),
    },
  };
  const comp = shallow(<MuiListItem {...props} />);
  return { comp, props };
}

describe('<MuiListItem />', () => {
  it('renders MuiListItem', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  it('handleClick and calls history push ', async () => {
    const { comp, props } = setup(true, '/page-test');
    comp.find('WithStyles(ForwardRef(ListItem))').at(0).simulate('click');
    expect(props.history.push).toBeCalled();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create((
      <MemoryRouter>
        <MuiListItem {...props} />
      </MemoryRouter>
    ));
    expect(tree).toMatchSnapshot();
  });
});
