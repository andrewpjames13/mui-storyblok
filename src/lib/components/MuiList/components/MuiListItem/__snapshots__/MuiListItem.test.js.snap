// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiListItem /> snapshot 1`] = `
<li
  className="MuiListItem-root makeStyles-root-4 MuiListItem-gutters"
  disabled={false}
  onClick={[Function]}
>
  <div
    className="MuiListItemAvatar-root makeStyles-root-5"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
    >
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root makeStyles-root-6 MuiIcon-colorSecondary"
      >
        android
      </span>
    </div>
  </div>
  <div
    className="MuiListItemIcon-root makeStyles-root-7"
  >
    <span
      aria-hidden={true}
      className="material-icons MuiIcon-root makeStyles-root-8 MuiIcon-colorSecondary"
    >
      android
    </span>
  </div>
  <div
    className="MuiListItemText-root makeStyles-root-9 MuiListItemText-multiline"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
    >
      primary
    </span>
    <p
      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
    >
      
    </p>
  </div>
  <div
    className="MuiListItemSecondaryAction-root makeStyles-root-10"
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root makeStyles-root-11"
      data-testid="muiIconButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <span
          aria-hidden={true}
          className="material-icons MuiIcon-root makeStyles-root-12 MuiIcon-colorSecondary"
        >
          star
        </span>
      </span>
    </button>
  </div>
</li>
`;
